src/censys/cloud_connectors/azure_connector/connector.py

```diff
@@ -56,72 +71,148 @@ def __init__(self, settings: Settings):
 
-    def get_all_labels(self):
+    def get_all_labels(self, subscription_id: str, label_prefix: str) -> set[str]:
         """Get Azure labels."""
         subscription_client = SubscriptionClient(self.credentials)
 
-        locations = subscription_client.subscriptions.list_locations(
-            self.subscription_id
-        )
+        locations = subscription_client.subscriptions.list_locations(subscription_id)
 
-        self.possible_labels.clear()
+        # self.possible_labels.clear()
+        possible_labels = set()
 
         for location in locations:
-            self.possible_labels.add(
-                f"{self.label_prefix}{self.subscription_id}/{location.name}"
+            # self.possible_labels.add(
+            possible_labels.add(
+                # f"{self.label_prefix}{self.subscription_id}/{location.name}"
+                f"{label_prefix}{subscription_id}/{location.name}"
             )
 
-    def scan(self):
-        """Scan Azure Subscription."""
+        return possible_labels
+
+    def scan(self, **kwargs):
+        """Scan Azure Subscription.
+
+        Args:
+            **kwargs: Keyword arguments.
+                scan_context (AzureScanContext): Azure scan context.
+        """
+        ctx: AzureScanContext = kwargs["scan_context"]
+
+        ctx.credentials = ClientSecretCredential(
+            tenant_id=ctx.provider_settings.tenant_id,
+            client_id=ctx.provider_settings.client_id,
+            client_secret=ctx.provider_settings.client_secret,
+        )
+
+        logger = get_logger(
+            log_name=f"{self.provider.lower()}_cloud_connector",
+            level=self.settings.logging_level,
+            provider=f"{self.provider}_{ctx.subscription_id}",
+        )
+        ctx.logger = logger
+        ctx.logger.info(f"Scanning {self.provider} - sub:{ctx.subscription_id}")
+
         with Healthcheck(
             self.settings,
-            self.provider_settings,
-            provider={"subscription_id": self.subscription_id},
+            ctx.provider_settings,  # self.provider_settings,
+            # provider={"subscription_id": self.subscription_id},
+            provider={"subscription_id": ctx.subscription_id},
             exception_map={
                 ClientAuthenticationError: "PERMISSIONS",
             },
         ):
-            super().scan()
+            super().scan(**kwargs)
 
     def scan_all(self):
         """Scan all Azure Subscriptions."""
         provider_settings: dict[
             tuple, AzureSpecificSettings
         ] = self.settings.providers.get(self.provider, {})
+
+        self.logger.debug(
+            f"scanning {self.provider} using {self.settings.scan_concurrency} processes"
+        )
+
+        label_prefix = self.get_provider_label_prefix()
+        pool = Pool(processes=self.settings.scan_concurrency)
+
         for provider_setting in provider_settings.values():
+            # this is so confusing - plural settings to setting?
```

Comment:
maybe `for azure_settings in provider_settings.values()`?
or `for azure_entry in provider_settings.values()`?
or `for provider_entry in provider_settings.values()`?

Reply:
I think changing the context property name from provider_settings to provider_setting would make the most sense. It's not a big deal, but I tripped on it a couple of times. AWS and GCP are probably doing the same.